enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md

```diff
@@ -0,0 +1,509 @@
+  fashion so that customer can use those operators as easily as possible, driving the use of layered products.
+
+### Goals
+
+Allow Day-2 Operators to access cloud provider resources as seamlessly as possible when being installed, used and
+updated on STS enabled clusters.
+
+While providing the above goal, allow for multi-tenancy. In this sense multi-tenancy means that an operator may
+provide different cloud provider credentials per operand such that operator users may uniquely access non-shared cloud
+resources.
+
+Operator authors have a way to notify, guide, and assist Day-2 Operator admins in providing the required cloud provider
+credentials matched to their permission needs for install and update.
+
+Ideally, a solution here will work in both HyperShift (STS always) and non-HyperShift clusters.
+
+### Non-Goals
+
+Day 1 operators are not included in this proposal, they are pre-provisioned for STS enabled clusters (at least 
+for HyperShift)
+
+
+
+## Proposal
+
+**Cloud Credential Operator (CCO) changes**: Adds a token-aware mode, while nominally set in "Manual" mode, CCO will
+look for and process CredentialRequests referenced in Operator CRs when:
+
+```bash
+# infrastructure platform is AWS
+$ oc get infrastructure cluster -o jsonpath --template '{ .status.platform }'
+AWS
+
+# credentialsMode is "Manual"
+$ oc get cloudcredential cluster -o jsonpath --template '{ .spec.credentialsMode }'
+Manual
+
+# serviceAccountIssuer is non empty
+$ oc get authentication cluster -o jsonpath --template='{ .spec.serviceAccountIssuer }'
+abutcher-oidc.s3.us-east-1.amazonaws.com
+```
+
+This will operate in one of two ways depending on which other
+parts of this EP for other components are adopted: specifically HyperShift/Rosa adding CCO with these token-aware
+changes or continuing with the current support for the pod identity webhook.
+
+CCO's token-aware mode will work by adding a "role-arn"-type field on operator added CredentialsRequest objects.
+This is a new API field on the CredentialsRequest spec. When CCO acquires a CredentialsRequest it will create a Secret 
+(as today mostly) with new functionality: Adding to the Secret a path to the projected ServiceAccount token 
+(static for OCP) and a roleARN (from the new field on the CredentialsRequest).
+
+Validation of CredentialsRequest by this new token-aware CCO? Maybe. Currently, CCO gets some permissions under ROSA
+(unused), could expand these permissions to include tools like AWS's Policy Simulator such that it could validate a role
+has permissions, and with this CCO could be the alerting mechanism for a changed CredentialsRequest without sufficient 
+permissions.
+
+**HyperShift changes**: Include Cloud Credential Operator with token-aware mode (see above). Allows for processing of 
+CredentialsRequest objects added by operators.
+
+**OperatorHub and Console changes**: Allow for import of additional, well-known, bundle annotations which will result 
```

Comment:
> **OperatorHub and Console changes**
To clarify, the proposed UX is purely client driven?
- Client reads the bundle metadata
- Client reads the cluster mode
- Client creates a Subscription and populates its env overrides accordingly?

Reply:
Seems like the priority is console users. There is precedent for this with multi arch for example where console users get a filter that CLI users don't get.